relay/cache/cache_accessor.go

```diff
@@ -74,11 +90,30 @@ func NewCacheAccessor[K comparable, V any](
 		concurrencyLimiter = make(chan struct{}, concurrencyLimit)
 	}
 
+	insertionTimes := make(map[K]time.Time)
+	var evictionHandler func(K, V)
+	if metrics != nil {
+		// If metrics are enabled, track the amount of time each item spends in the cache.
+		// Thread safety is provided by the cacheLock.
+		evictionHandler = func(key K, _ V) {
+			if insertionTime, ok := insertionTimes[key]; ok {
+				lifespan := time.Since(insertionTime).Milliseconds()
+				metrics.lifespan.Set(float64(lifespan))
+				delete(insertionTimes, key)
+			}
+		}
+	}
+
+	cache := NewFIFOCache(maxWeight, calculator, evictionHandler)
```

Comment:
why do we now create this cache from the constructor vs. taking it as a parameter?

Reply:
This is necessary due to the new `evictionHandler`, which is used to determine how long elements are in the cache before they are evicted. A few options if you'd like to back to the prior pattern:

- remove the metric that reports on the time each element remains in the cache (I think this will be a useful metric, would like to figure out how to keep it)
- add a method `Cache.SetEvictionHandler(handler)` that allows an eviction callback to be registered after the cache is constructed
- use a builder pattern, i.e. where there is a `CacheBuilder` class that you can configure by calling optional setters that eventually builds the cache